compaction/compaction.cc

```diff
@@ -1242,15 +1251,25 @@ class scrub_compaction final : public regular_compaction {
         flat_mutation_reader_v2 _reader;
         mutation_fragment_stream_validator _validator;
         bool _skip_to_next_partition = false;
+        compaction_task_stats_ptr _cts;
 
     private:
+        inline void register_validation_error() {
+            ++_cts->validation_errors;
+        }
+
+        inline void register_validation_error_fix() {
+            ++_cts->fixed_validation_errors;
+        }
+
         void maybe_abort_scrub() {
             if (_scrub_mode == compaction_type_options::scrub::mode::abort) {
                 throw compaction_aborted_exception(_schema->ks_name(), _schema->cf_name(), "scrub compaction found invalid data");
             }
```

Comment:
I think this would be the best place to increment error metrics. You can bump `validation_errors` before the if and `fixed_validation_errors` after the if (if we don't abort, we fix the error). BTW why count fixes too? When we don't abort, we fix all errors we find. Don't really see the point in two separate metrics, they will always have the same value.